com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ModuleHooksAggregator.java

```diff
@@ -82,7 +82,26 @@ private boolean shouldExecuteHook(Map<String, List<String>> alreadyExecutedHooks
     private boolean hasAllPhasesExecuted(Map<String, List<String>> alreadyExecutedHooks, Hook hookToBeExecuted,
                                          List<HookPhase> hookPhasesForCurrentStepPhase) {
         List<HookPhase> executedHookPhasesForHook = getExecutedHookPhasesForHook(alreadyExecutedHooks, hookToBeExecuted.getName());
-        return executedHookPhasesForHook.containsAll(hookPhasesForCurrentStepPhase);
+        return executedHookPhasesForHook.containsAll(getHookPhasesForCurrentStepPhase(hookToBeExecuted, hookPhasesForCurrentStepPhase));
```

Comment:
extract this getHookPhasesForCurrentStepPhase(hookToBeExecuted, hookPhasesForCurrentStepPhase) to a variable